routes/shopItem.js

```diff
@@ -7,10 +7,18 @@ shopItemRouter = express.Router();
 
 shopItemRouter.post("/", (req, res, next) => {
     try {
-        id++;
-        req.body.id = id;
-        fs.writeFileSync(resolve("./database.json"), JSON.stringify(req.body) + "\n", { flag: "a+" });
-        res.json(`${req.method} success`);
+        if (typeof(req.body.name) === "string" && typeof(req.body.isCompleted === "boolean")) {
```

Comment:
`typeof(req.body.isCompleted === "boolean")` is not what you want since that will return you the type of `req.body.isCompleted === "boolean"` expression

```suggestion
        if (typeof(req.body.name) === "string" && typeof(req.body.isCompleted) === "boolean") {
```